primitives/consensus/babe/src/digests.rs

```diff
@@ -18,178 +18,80 @@
 
 #[cfg(feature = "std")]
 use super::{BABE_ENGINE_ID, AuthoritySignature};
-#[cfg(not(feature = "std"))]
-use super::{VRF_OUTPUT_LENGTH, VRF_PROOF_LENGTH};
 use super::{AuthorityId, AuthorityIndex, SlotNumber, BabeAuthorityWeight};
+use sp_core::RuntimeDebug;
 #[cfg(feature = "std")]
 use sp_runtime::{DigestItem, generic::OpaqueDigestItemId};
 #[cfg(feature = "std")]
 use std::fmt::Debug;
 use codec::{Decode, Encode};
 #[cfg(feature = "std")]
-use codec::{Codec, Input, Error};
-#[cfg(feature = "std")]
-use schnorrkel::{
-	SignatureError, errors::MultiSignatureStage,
-	vrf::{VRFProof, VRFOutput, VRF_OUTPUT_LENGTH, VRF_PROOF_LENGTH}
-};
+use codec::Codec;
+use sp_consensus_vrf::schnorrkel::{VRFProof, VRFOutput, Randomness};
 use sp_std::vec::Vec;
 
+/// A BABE primary pre-runtime digest;
+#[derive(Clone, RuntimeDebug, Encode, Decode)]
+pub struct PrimaryPreDigest {
+	/// VRF output
+	pub vrf_output: VRFOutput,
+	/// VRF proof
+	pub vrf_proof: VRFProof,
+	/// Authority index
+	pub authority_index: AuthorityIndex,
+	/// Slot number
+	pub slot_number: SlotNumber,
+}
+
+/// A BABE secondary pre-runtime digest.
+#[derive(Clone, RuntimeDebug, Encode, Decode)]
+pub struct SecondaryPreDigest {
+	/// Authority index
+	pub authority_index: AuthorityIndex,
+	/// Slot number
+	pub slot_number: SlotNumber,
+}
 
 /// A BABE pre-runtime digest. This contains all data required to validate a
 /// block and for the BABE runtime module. Slots can be assigned to a primary
 /// (VRF based) and to a secondary (slot number based).
-#[cfg(feature = "std")]
-#[derive(Clone, Debug)]
+#[derive(Clone, RuntimeDebug, Encode, Decode)]
```

Comment:
Is the binary encoding still the same compared to `RawPreDigest`? I think it is the same but just want to double check.

Reply:
Yeah it's the same. The only difference of `RawPreDigest` and `PreDigest` was `VRFProof` and `VRFOutput`. The wrapped `VRFProof` and `VRFOutput` always encode/decode with the same format as `RawVRFProof` and `RawVRFOutput`.